pkg/storage/tsdb/block/block.go

```diff
@@ -121,6 +121,14 @@ func Upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, blockDi
 		return cleanUp(logger, bkt, id, errors.Wrap(err, "upload index"))
 	}
 
+	if err := objstore.UploadFile(ctx, logger, bkt,
+		filepath.Join(blockDir, SparseIndexHeaderFilename),
+		filepath.Join(id.String(), SparseIndexHeaderFilename),
+	); err != nil {
+		// Don't call cleanUp. Uploading sparse index headers is best effort.
+		level.Warn(logger).Log("msg", "failed to upload sparse index headers", "block", id.String(), "err", err)
+	}
```

Comment:
this would log a warning even if the config option to build sparser headers is disabled. Can we not attempt to upload the file if it's not on the local fs?